lxd/storage/backend_lxd.go

```diff
@@ -2119,92 +2121,198 @@ func (b *lxdBackend) CreateInstanceFromMigration(inst instance.Instance, conn io
 	args.Config = vol.Config()
 	args.Name = inst.Name()
 
-	projectName := inst.Project().Name
+	if len(args.ConvertOptions) == 0 {
+		projectName := inst.Project().Name
 
-	// If migration header supplies a volume size, then use that as block volume size instead of pool default.
-	// This way if the volume being received is larger than the pool default size, the block volume created
-	// will still be able to accommodate it.
-	if args.VolumeSize > 0 && contentType == drivers.ContentTypeBlock {
-		b.logger.Debug("Setting volume size from offer header", logger.Ctx{"size": args.VolumeSize})
-		args.Config["size"] = fmt.Sprintf("%d", args.VolumeSize)
-	} else if args.Config["size"] != "" {
-		b.logger.Debug("Using volume size from root disk config", logger.Ctx{"size": args.Config["size"]})
-	}
+		// If migration header supplies a volume size, then use that as block volume size instead of pool default.
+		// This way if the volume being received is larger than the pool default size, the block volume created
+		// will still be able to accommodate it.
+		if args.VolumeSize > 0 && contentType == drivers.ContentTypeBlock {
+			b.logger.Debug("Setting volume size from offer header", logger.Ctx{"size": args.VolumeSize})
```

Comment:
There's a contextual logger inside this function already defined at the top as `l` we should use it rather than using the pool level logger and losing context of the request.